File: src/main/java/github/stephenwanjala/travelagency/dao/BookingDAO.java
package github.stephenwanjala.travelagency.dao;

import github.stephenwanjala.travelagency.model.Booking;

import java.util.List;

public interface BookingDAO {
    List<Booking> getAllBookings();
    Booking getBookingById(int id);
    void addBooking(Booking booking);
    void deleteBooking(int id);
}

File: src/main/kotlin/data/dao/BookingDao.kt
package data.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Query
import androidx.room.Upsert

@Dao
interface BookingDao {

    @Query("SELECT * FROM Booking")
    suspend fun getAllBookings(): List<models.Booking>

    @Query("SELECT * FROM Booking WHERE id = :id")
    suspend fun getBookingById(id: Int): models.Booking

    @Upsert
    suspend fun insertBooking(booking: models.Booking)

    @Delete
    suspend fun deleteBooking(booking: models.Booking)


}